### PR TITLE
Bug fix : 363595, 363659,365989, Configuring Testagent as interactive process in HTTPs

### PR DESCRIPTION
**Pull request by aditya-MSFT**
1. Making sure we are deleting the DTAService ahead of configuring the agent again
2. Making sure we are launching PSSession with proper port and protocol. There is an active bug against DTL 366007 to provide back the private port for VMs, 
3. Making sure we are using correct DateTime format while scheduling the task
4. Making sure we are returning correct code for rebooting the machine

Testing:
1. Azure : Interactive, Service, fresh installation, existing environment
2. OnPrem: Fresh Installation , existing installation, as interactive, as process
3. With Codecoverage 
4. JPN test agent machine
5. Configuring as Non Admin

### CODE REVIEW

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 1]
```diff
@@ -606,13 +630,62 @@ function InvokeDTAExecHostExe([string] $Version, [System.Management.Automation.P
...
+    Try
+    {
+        $StartDate = New-Object -TypeName DateTime -ArgumentList:(2050,01,01)
+        $FormatHack = ($([System.Globalization.DateTimeFormatInfo]::CurrentInfo.ShortDatePattern) -replace 'M+/', 'MM/') -replace 'd+/', 'dd/'
+        $date1 = $StartDate.ToString($FormatHack)
 
-    $session = New-PSSession -ComputerName . -Credential $MachineCredential
-    Invoke-Command -Session $session -ErrorAction Continue -ErrorVariable err -OutVariable out -scriptBlock { schtasks.exe /create /TN:DTAConfig /TR:$args /F /RL:HIGHEST /SD:01/01/2050 /SC:ONCE /ST:00:00 ; schtasks.exe /run /TN:DTAConfig } -ArgumentList $exePath
+        $session = CreateNewSession -MachineCredential $MachineCredential
+        Invoke-Command -Session $session -ErrorAction Continue -ErrorVariable err -OutVariable out -scriptBlock { schtasks.exe /create /TN:DTAConfig /TR:$args[0] /F /RL:HIGHEST /SD $args[1] /SC:ONCE /ST:00:00 ; schtasks.exe /run /TN:DTAConfig } -ArgumentList $exePath,$date1
```
Comment (kmkumaran) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: https://msdn.microsoft.com/en-us/library/windows/desktop/bb736357(v=vs.85).aspx says /SD option is not applicable for scheduling type ONCE. Any reason why we have to pass this?

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 2]
```diff
@@ -606,13 +630,62 @@ function InvokeDTAExecHostExe([string] $Version, [System.Management.Automation.P
...
+{
+    Write-Verbose -Message("Trying to fetch WinRM details on the machine") -Verbose
 
-    Write-Verbose ("Error : {0} " -f ($err | out-string)) -Verbose
-    Write-Verbose ("Output : {0} " -f ($out | out-string)) -Verbose
+    $winrmconfigDetails = Winrm e  winrm/config/listener -format:pretty |Out-String
+    $xmldoc = [XML]$winrmconfigDetails
+    $ns = new-object Xml.XmlNameSpaceManager $xmldoc.NameTable
+    $ns.AddNameSpace("cfg","http://schemas.microsoft.com/wbem/wsman/1/config/listener")
+    $ns.AddNameSpace("xsi","http://www.w3.org/2001/XMLSchema-instance")
```
Comment (kmkumaran) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Is this namespace same across all OS that we support?

Comment (aditya-MSFT) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Yes, I have validated it across all supported OS.

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 1 above)
Comment (aditya-MSFT) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: I didn't investigate which all parameters are mandatory and which are not, but thanks for comment, will revisit the current list and see if we can avoid it completely.

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 3]
```diff
@@ -714,7 +787,7 @@ function ConfigureTestAgent
         throw ("TestAgent Configuration failed with exit code {0}. Error code : {1}" -f $LASTEXITCODE, $retCode)
     }
 
-    return $ret;
+    return $retCode;
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: I feel this fix (retcode) shuld go in m85.

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 4]
```diff
@@ -606,13 +630,62 @@ function InvokeDTAExecHostExe([string] $Version, [System.Management.Automation.P
...
+        Write-Verbose ("Error : {0} " -f ($err | out-string)) -Verbose
+        Write-Verbose ("Output : {0} " -f ($out | out-string)) -Verbose
+    }
+    Catch [Exception]
+    {
+        Write-Verbose -Message ("Unable to start Agent process, will be rebooting the machine to complete the configuration {0}" -f  $_.Exception.Message) -Verbose
+    }
+}
+
+function CreateNewSession( [System.Management.Automation.PSCredential] $MachineCredentials)
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: write todo with user story / bug id

**Comment by acesiddhu** [comment]
signed off

**Comment by kmkumaran** [comment]
Looks fine.

**Pull request merged**